package.json

```diff
@@ -803,12 +803,17 @@
       }
     ]
   },
-  "main": "./out/client/src/extension",
+  "main": "./client/out/extension",
   "scripts": {
-    "vscode:prepublish": "npm run compile",
-    "compile": "webpack && tsc -b",
-    "watch": "tsc -b -w",
-    "test": "node ./node_modules/vscode/bin/test"
+    "vscode:prepublish": "npm run package",
+    "package": "webpack --mode production --config webpack.config.js",
+    "package:dev": "webpack --mode development --config webpack.config.js",
+    "compile": "webpack --mode none --config webpack.config.js",
+    "compile:client": "webpack --mode none --config webpack.client.config.js",
+    "compile:html_views": "webpack --mode none --config ./webpack.html_views.config.js",
+    "compile:server": "webpack --mode none --config ./webpack.server.config.js",
+    "watch": "webpack --watch --mode none --config webpack.config.js",
```

Comment:
I would drop "compile:server", "compile:client", and "compile:html_views", as webpack is smart enough to compile only what's necessary: a `webpack` invoked from the toplevel directory should do whatever is necessary, and it's unlikely that any developer would invoke any of these three build targets individually.

Further, `--config webpack.config.js` can be stripped, as webpack automatically looks for a `webpack.config.js` by default. Perhaps add `--devtool hidden-source-map` to "package", and drop the extraneous `--mode none`, as the common config specifies this mode.

Finally, I might be tempted to drop the "package:dev" target, as it's unlikely to be used directly.